BenefitsAllocationUpload/Controllers/ReportsController.cs

```diff
@@ -32,11 +33,27 @@ public ReportsController(IRepository<UnitFile> unitFileRepository, ISftpService
             _unitFileRepository = unitFileRepository;
         }
 
-        private FileNames GetNamedFile(string id)
+        /// <summary>
+        /// Returns a file when the _objData is known.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        private FileNames GetFile(string id)
         {
             int fid = Convert.ToInt32(id);
             var files = _objData.GetFiles();
             return (files.Where(f => f.FileId == fid)).First();
+        }
+
+        /// <summary>
+        /// Tries to return a file with the given filename.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns></returns>
+        private FileNames GetFileByName(string name)
+        {
+            var files = _objData.GetFiles();
+            return (files.Where(f => f.FileName.Equals(name))).FirstOrDefault();
```

Comment:
@taylorkj i think you can replace this whole thing with `files.FirstOrDefault(f=>f.FileName == name)`